noxfile.py

```diff
@@ -0,0 +1,263 @@
+    list_packages(session)
+    run_pytest(session)
+
+
+@nox.session(venv_backend="conda", python=PYTHON_VERSIONS)
+def test_conda(session):
+    """
+    Run the tests and measure coverage (using conda)
+    """
+    install_requirements(session, ["build", "run", "test"], package_manager="conda")
+    build_project(session, install=True)
+    list_packages(session, package_manager="conda")
+    run_pytest(session)
+
+
+@nox.session()
+def docs(session):
+    """
+    Build the documentation
+    """
+    install_requirements(session, ["build", "run", "docs"])
+    build_project(session, install=True)
+    list_packages(session, package_manager="pip")
+    # Generate the API reference
+    session.run(
+        "sphinx-autogen",
+        "-i",
+        "-t",
+        str(DOCS / "_templates"),
+        "-o",
+        str(DOCS / "api" / "generated"),
+        str(DOCS / "api" / "index.rst"),
+    )
+    # Build the HTML pages
+    session.run(
+        "sphinx-build",
+        "-d",
+        str(DOCS / "_build" / "doctrees"),
+        "doc",
+        str(DOCS / "_build" / "html"),
+    )
+
+
+@nox.session()
+def build(session):
+    """
+    Build source and wheel distributions for the project
+    """
+    install_requirements(session, ["build"])
+    packages = build_project(session, install=False)
+    check_packages(session, packages)
+    list_packages(session)
+
+
+@nox.session
+def clean(session):
+    """
+    Remove all files generated by the build process
+    """
+    files = [
+        Path("build"),
+        Path("dist"),
+        Path("MANIFEST"),
+        Path(".coverage"),
+        Path(".pytest_cache"),
+        Path("__pycache__"),
+        DOCS / "_build",
+        DOCS / "api" / "generated",
+        DOCS / "gallery",
+        DOCS / "tutorials",
+    ]
+    files.extend(Path(PACKAGE).glob("**/*.pyc"))
+    files.extend(Path(".").glob(".coverage.*"))
+    files.extend(Path(PACKAGE).glob("**/__pycache__"))
+    files.extend(Path(".").glob("*.egg-info"))
+    files.extend(Path(".").glob("**/.ipynb_checkpoints"))
+    for f in files:
+        if f.exists():
+            session.log(f"removing: {f}")
+            if f.is_dir():
+                shutil.rmtree(f)
+            else:
+                f.unlink()
+
+
+# UTILITY FUNCTIONS
+###############################################################################
+
+
+def install_requirements(session, requirements, package_manager="pip"):
+    """
+    Install dependencies from the requirements files specified in REQUIREMENTS.
+
+    *requirements* should be a list of keywords defined in the REQUIREMENTS
+    dictionary. Set *package_manager* to "conda" if using the conda backend for
+    virtual environments.
+    """
+    if package_manager not in {"pip", "conda"}:
+        raise ValueError(f"Invalid package manager '{package_manager}'")
+    if session.posargs and "skip-install" in session.posargs:
+        session.log(f"Skipping install steps.")
```

Comment:
We don't need to use f-string here.
```suggestion
        session.log("Skipping install steps.")
```